### PR TITLE
Fix flaky Macvtap test: Increase macvtap interface IP report timeout

### PR DESCRIPTION
**Pull request by ormergi**
Signed-off-by: Or Mergi <ormergi@redhat.com>

**What this PR does / why we need it**:
Following #6318 

We see macvtap test failures due to timeout of reporting the macvtap interface IP address
for a while now [1] [2] [3].
This PR increase the time we wait for it to be reported.

[1] https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6267/pull-kubevirt-e2e-k8s-1.21-sig-network/1432264444922040320
[2] https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6312/pull-kubevirt-e2e-k8s-1.21-sig-network/1432303029054345216
[3] https://search.ci.kubevirt.io/?search=should+have+managed+to+figure+out+the+IP+of+the+server+VMI&maxAge=336h&context=1&type=all&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubevirt/kubevirt/issues/6500
**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by maiqueb — COMMENTED** [review]
I think waiting 4 minutes for a vm to come up is a lot of time. 

And you're doubling the current timeout - why doubling ?

What causes the VM bootup to take longer than before ? This is the question I'd like to see the answer to, before accepting doubling the timeout without knowing why.

It could even be preferable to just replace this problem with another: use static addresses also for the migration test. That way, we don't need to wait for the address to be reported (but we'd need to properly configure the macvtap interface, instead of relying on the DHCP server configuring the host's network.

**Comment by ormergi** [comment]
@maiqueb 
> I think waiting 4 minutes for a vm to come up is a lot of time.
> 
> And you're doubling the current timeout - why doubling ?
> 

I doubled it in order to give QEMU guest agent reasonable time to start and report.

> What causes the VM bootup to take longer than before ? This is the question I'd like to see the answer to, before accepting doubling the timeout without knowing why.
> 

I think we have this issue since day one, it just that it occurs not often.
From what I understand and saw the root cause is that the macvtap interface IP address takes more time then we though to be reflected on the VMI status.

> It could even be preferable to just replace this problem with another: use static addresses also for the migration test. That way, we don't need to wait for the address to be reported (but we'd need to properly configure the macvtap interface, instead of relying on the DHCP server configuring the host's network).

Why you consider this as replacing a problem with another?
Configuring the macvtap iface could be done with exec'ing the VMI right?

**Comment by maiqueb** [comment]
> @maiqueb
> 
> > I think waiting 4 minutes for a vm to come up is a lot of time.
> > And you're doubling the current timeout - why doubling ?
> 
> I doubled it in order to give QEMU guest agent reasonable time to start and report.

Yes, but how do you define "reasonable" ? I'm just trying to point out that we do not understand the reasons why this takes longer, nor where the extra time is spent.

> 
> > What causes the VM bootup to take longer than before ? This is the question I'd like to see the answer to, before accepting doubling the timeout without knowing why.
> 
> I think we have this issue since day one, it just that it occurs not often. From what I understand and saw the root cause is that the macvtap interface IP address takes more time then we though to be reflected on the VMI status.

Yes, but where is that extra time spent ? What guarantees do we have that in 3 months you're not pushing another PR bumping from 4 to 8 minutes ? 

Again, I'm just trying to say that we do not understand this problem. And you're not fixing it; you're just relaxing our time expectations of what "reasonable" is. I have serious doubts that 4 minutes for a VM to be available is reasonable.

> 
> > It could even be preferable to just replace this problem with another: use static addresses also for the migration test. That way, we don't need to wait for the address to be reported (but we'd need to properly configure the macvtap interface, instead of relying on the DHCP server configuring the host's network).
> 
> Why you consider this as replacing a problem with another? Configuring the macvtap iface could be done with exec'ing the VMI right?

Because you have to properly configure the VMI, while before you get that for free since it the node's network DHCP server doing it for you. 

I'd prefer to have it over some kind of API, but, I'm OK if it starts with exec'ing into the VM. Once we have that, we can then improve it.

File: tests/network/vmi_multus.go [hunk 1]
```diff
@@ -1310,9 +1310,11 @@ var _ = SIGDescribe("[Serial]Macvtap", func() {
 			var serverVMIPodName string
 			var serverIP string
 
-			getVMMacvtapIfaceIP := func(vmi *v1.VirtualMachineInstance, macAddress string) (string, error) {
+			macvtapIfaceIPReportTimeout := 4 * time.Minute
```
[the remark this answers is not in the document]
Comment (AlonaKaplan) [reply tests/network/vmi_multus.go:1313]: I"m ok with not waiting for this intermediate state (although other tests with ga wait for it and I find it useful to be aligned with the other tests), but somebody decided that it is ok to wait 12 min for the agent to be connected, why here 4 min is enough?

Just to be clear, I"m not saying you have to increase the timeout to 12 min:)

**Pull request merged**